src/hypertable.c

```diff
@@ -1319,14 +1319,17 @@ hypertable_validate_constraints(Oid relid)
 	{
 		Form_pg_constraint form = (Form_pg_constraint) GETSTRUCT(tuple);
 
-		if (form->contype == CONSTRAINT_FOREIGN)
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_TABLE_DEFINITION),
-					 errmsg("cannot have FOREIGN KEY constraints to hypertable \"%s\"",
-							get_rel_name(relid)),
-					 errhint("Remove all FOREIGN KEY constraints to table \"%s\" before "
-							 "making it a hypertable.",
-							 get_rel_name(relid))));
+    /*
+     * Hypertable <-> hypertable foreign keys are not supported.
+     */
+	  if (form->contype == CONSTRAINT_FOREIGN && ts_hypertable_relid_to_id(form->conrelid) != -1)
```

Comment:
Why not use INVALID_HYPERTABLE_ID here?

Reply:
Because the function returns -1 if this is not a hypertable and INVALID_HYPERTABLE_ID is 0

Reply:
Then the function should return INVALID_HYPERTABLE_ID instead of -1 to be in sync with the rest of the code base.